idseq_dag/engine/pipeline_step.py

```diff
@@ -49,6 +50,18 @@ def run(self):
     def count_reads(self):
         ''' count reads '''
 
+    def count_reads_work(self):
```

Comment:
Now that it has a default implementation, we do not need the above abstract method, right?
Just override this method when necessary and call super.count_reads_works to reference the parent method.

Reply:
@yunfangjuan had a preference for keeping it an abstract method -- let's discuss. If we do remove @abstractmethod, I think there is an argument to be made that the default method should still be "pass": most steps added in the future will not need counting.

Reply:
ok, I might not have enough context. 
Abstract methods normally MUST be implemented. On the other hand a simple method with default pass provide a hook that subclass can use to execute custom code.
As I said, I do not have enough context, but jut by looking at these particular files it seemed the best option.